parametrize fails on null string in 8.0.dev

<!--
Thanks for submitting an issue!

Quick check-list while reporting bugs:
-->

This was originally reported at https://github.com/pytest-dev/pytest/pull/11489#issuecomment-1783376878 but was asked to open a new issue for it. Also see:

* https://github.com/astropy/astropy/pull/15548

Caused by:

* #11489

https://github.com/pytest-dev/pytest/blob/ed8701a57fa40296e8dd302015cd5f805f2aede0/src/_pytest/python.py#L1006

To reproduce the failure, create a test using parametrize marker and include empty string as one of the parametrize inputs. Example traceback:

```
_ ERROR collecting astropy/time/tests/test_mask.py _
...
_pytest/python.py:1338: in parametrize
    ids = self._resolve_parameter_set_ids(
_pytest/python.py:1461: in _resolve_parameter_set_ids
    return id_maker.make_unique_parameterset_ids()
_pytest/python.py:1006: in make_unique_parameterset_ids
    if id[-1].isdigit():
E   IndexError: string index out of range
```

--- a/changelog/11563.bugfix.rst
+++ b/changelog/11563.bugfix.rst
@@ -1,0 +1,1 @@
+Fixed crash when using an empty string for the same parametrized value more than once.

--- a/src/_pytest/python.py
+++ b/src/_pytest/python.py
@@ -1003,7 +1003,7 @@ class IdMaker:
             for index, id in enumerate(resolved_ids):
                 if id_counts[id] > 1:
                     suffix = ""
-                    if id[-1].isdigit():
+                    if id and id[-1].isdigit():
                         suffix = "_"
                     new_id = f"{id}{suffix}{id_suffixes[id]}"
                     while new_id in set(resolved_ids):

--- a/testing/python/metafunc.py
+++ b/testing/python/metafunc.py
@@ -626,6 +626,13 @@ class TestMetafunc:
             ).make_unique_parameterset_ids()
             assert result == [expected]
 
+    def test_idmaker_duplicated_empty_str(self) -> None:
+        """Regression test for empty strings parametrized more than once (#11563)."""
+        result = IdMaker(
+            ("a",), [pytest.param(""), pytest.param("")], None, None, None, None, None
+        ).make_unique_parameterset_ids()
+        assert result == ["0", "1"]
+
     def test_parametrize_ids_exception(self, pytester: Pytester) -> None:
         """
         :param pytester: the instance of Pytester class, a temporary

